File: backend/internal/models/event.go
package models

import "time"

const (
	EventAvailable = "available"
	EventSoldOut   = "sold_out"
)

type Event struct {
	ID             uint      `gorm:"primaryKey" json:"id"`
	Title          string    `json:"title"`
	Description    string    `json:"description"`
	Location       string    `json:"location"`
	EventDate      time.Time `json:"event_date"`
	EventTime      string    `json:"event_time"`
	Category       string    `json:"category"`
	TotalSeats     int       `json:"total_seats"`
	AvailableSeats int       `json:"available_seats"`
	Capacity       int       `json:"capacity"`
	Price          float64   `json:"price"`
	Organizer      string    `json:"organizer"`
	BannerURL      string    `json:"banner_url"`
	Status         string    `json:"status"`
	CreatedAt      time.Time `json:"created_at"`
	UpdatedAt      time.Time `json:"updated_at"`

	Bookings []Booking `json:"bookings"`
}

File: frontend/types/event.ts
export interface Seat {
  id: number
  event_id: number
  seat_number: string
  is_booked: boolean
  booking_id?: number
}

export interface Event {
  id: number
  title: string
  description: string
  location: string
  event_date: string
  event_time: string
  duration_mins: number
  category: string
  total_seats: number
  available_seats: number
  capacity: number
  price: number
  organizer: string
  banner_url: string
  status: string
  created_at: string
  updated_at: string
}